modules/woo-commerce/woo-commerce.php

```diff
@@ -42,3 +42,63 @@ function qtranxf_wc_detect_language( $url_info ) {
 }
 
 add_filter( 'qtranslate_detect_language', 'qtranxf_wc_detect_language', 5 );
+
+/**
+ * Handler for webhooks, which should always send information in Raw ML format.
+ *
+ * For some cases (e.g. variations updates) the webhook is generated through AJAX instead of cron.
+ * In that context, qwc-admin.php is loaded instead of qwc-front.php
+ */
+function qtranxf_wc_deliver_webhook_async( $webhook_id, $arg ) {
+    if ( function_exists( 'qtranxf_get_front_page_config' ) ) {
+        $page_configs = qtranxf_get_front_page_config();
+        if ( ! empty( $page_configs['']['filters'] ) ) {
+            qtranxf_remove_filters( $page_configs['']['filters'] );
+        }
+    }
+
+    remove_filter( 'get_post_metadata', 'qtranxf_filter_postmeta', 5 );
+    remove_filter( 'the_posts', 'qtranxf_postsFilter', 5 );
+    remove_action( 'pre_get_posts', 'qtranxf_pre_get_posts', 99 );
+
+    /* Raw ML format is not applicable to terms, as default lang only is stored in obj->name and translations are in qtx options.
+     * Hence qtranxf_wc_get_term_raw_ML filter is added to mimic a raw ML format to be sent through webhook.
+     */
+    add_filter( 'get_term', 'qtranxf_wc_get_term_raw_ML' );
+    add_filter( 'get_terms', 'qtranxf_wc_get_term_raw_ML' );
+    wp_cache_flush();
+
+    /* Remove admin filters which can affect webhooks
+     * TODO: check if qtranxf_wc_add_filters_admin() can be called only if not doing cron or relevant filters are needed in other cron operations.
+     * In that case qtranxf_wc_add_filters_admin() can be called conditionally, following qtranxf_remove_filters call can be removed.
+     * Otherwise all filters affecting webhooks added in qtranxf_wc_add_filters_admin() must be removed here.
+     */
+    qtranxf_remove_filters( [
+        'text' => [
+            'woocommerce_attribute_taxonomies'  => 20,
+            'woocommerce_variation_option_name' => 20,
+        ]
+    ] );
+
```

Comment:
Great you came up with this fix. 
However this webhook is a bit difficult to maintain as it's removing many filters set earlier so we have to be sure we don't miss any. Would be better to detect the webhook case earlier, but I guess this `woocommerce_deliver_webhook_async` action is fired too late.

Reply:
> this webhook is a bit difficult to maintain

Definitely, but I did not find other ways, since as you guess the specific filter is coming too late to have a cleaner approach.
However we should take care mostly of the filters added in `qtranxf_wc_add_filters_admin()`, as `qtranxf_wc_add_filters_front()` is already prevented to be called if `wp_doing_cron()` is `true`.